internal/controller/suite_test.go

```diff
@@ -77,14 +85,30 @@ var _ = BeforeSuite(func() {
 
 	//+kubebuilder:scaffold:scheme
 
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
```

Comment:
Question, not a review, but why do we need manager here?

Reply:
I start the manager here because its a more complete integration test. I can incorporate the custom event handling logic for endpoint slices etc.